client/lib/ffi.rs

```diff
@@ -0,0 +1,126 @@
+//! Foreign function interfaces.
+
+use std::{
+    ffi::CStr,
+    os::raw::{c_char, c_uchar},
+    slice,
+    sync::Mutex,
+};
+
+use lazy_static::lazy_static;
+use tokio::runtime;
+
+use super::error::{Error, Result};
+
+lazy_static! {
+    static ref LAST_ERROR: Mutex<Option<Error>> = Mutex::new(None);
+    static ref RUNTIME: Mutex<runtime::Runtime> =
```

Comment:
I wonder if it might make sense to have explicit initialization and cleanup functions for the mutex-wrapped runtime in the C API.  

cc @xcthulhu

Reply:
Certainly doable, I added this in the last few moments just to get the example working.

Reply:
I've done this now, thanks.